### PR TITLE
feat: adds bookmarks sort

### PR DESCRIPTION
**Pull request by kendallstrautman**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-ksadds-bookmarks-sort-hashicorp.vercel.app/) 🔎
- [Asana task](https://app.asana.com/0/1199634971449915/1202832753754546) 🎟️

## 🗒️ What

This PR adds a sort feature to the tutorial card view. This sort occurs based on `created_date` for the bookmark. 

## 📸 Design Screenshots

Screen Shot 2022-08-25 at 1 05 03 PM

## 🧪 Testing

- [ ] pull down this branch, run things locally and navigate to /profile/bookmarks
- [ ] test the sort dropdown, verify that the card order reverses.

### CODE REVIEW

File: src/components/dropdown-disclosure/dropdown-disclosure.module.css [hunk 1]
```diff
@@ -23,7 +23,7 @@
 	padding-top: 4px;
 	position: absolute;
 	width: fit-content;
-	z-index: 1;
+	z-index: 2;
```
Comment (kendallstrautman) [inline src/components/dropdown-disclosure/dropdown-disclosure.module.css:26]: The disclosure was showing up behind the bookmark icon 🤷‍♀️

**Comment by kendallstrautman** [comment]
Just noticed the sort needs to be reversed I think, looking into it

File: src/components/dropdown-disclosure/components/activator/index.tsx [hunk 2]
```diff
@@ -33,7 +33,11 @@ const DropdownDisclosureActivator = ({
 			<Button
 				{...buttonProps}
 				color={color}
-				icon={hideChevron ? null : <IconChevronDown16 />}
+				icon={
+					hideChevron ? null : (
+						<IconChevronDown16 className={s.chevronWrapper} />
```
Comment (kendallstrautman) [inline src/components/dropdown-disclosure/components/activator/index.tsx:38]: The chevron animation wasn't occurring when the disclosure opened. Lmk if there's a different way to do this! I'm not sure of the context why the disclosure is particular about the child type being a string and using this.

File: src/components/dropdown-disclosure/dropdown-disclosure.module.css (hunk 1 above)
Comment (ashleemboyer) [reply src/components/dropdown-disclosure/dropdown-disclosure.module.css:26]: Oh good catch! This is definitely the way. 👍🏻

File: src/views/profile/bookmarks-view/helpers/card-sort-data.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+			return new Date(b.created_at).getTime() - new Date(a.created_at).getTime()
+		},
+	},
+	oldest: {
+		text: 'Oldest',
+		sort: (a: ApiBookmark, b: ApiBookmark) => {
+			return new Date(a.created_at).getTime() - new Date(b.created_at).getTime()
+		},
+	},
+}
```
Comment (ashleemboyer) [inline src/views/profile/bookmarks-view/helpers/card-sort-data.ts:16]: I really like this structure! It came about with the bookmark button as well. Very clear and consumable.

File: src/components/dropdown-disclosure/components/activator/index.tsx (hunk 2 above)
Comment (ashleemboyer) [reply src/components/dropdown-disclosure/components/activator/index.tsx:38]: I forgot about this!! It was a bug pointed out by a DS person recently and my brain forgot to make a ticket. :lolsob: This is exactly what needed to be done! 🎉

**Review by ashleemboyer — APPROVED** [review]
Looks great!! 🥇

**Pull request merged**